geometric/tests/addons.py

```diff
@@ -46,6 +46,12 @@ def get_gaussian_version():
 def workqueue_found():
     return (_plugin_import("work_queue") is True) and (geometric.nifty.which('work_queue_worker'))
 
+def bigchem_found(engine):
+    geometric.nifty.CheckBigChem(engine)
+    found = geometric.nifty.BigChemReady()
+    geometric.nifty.BigChemOff()
+    return found
+
```

Comment:
Again, you want to consider your system architecture and remove these sorts of global checks. Try implementing methods that now include bigchem with the `bigchem: bool = False` keyword argument. Having the default value `False` ensures that all former client code (in other people's code bases) the relied upon a signature _without_ the `bigchem` kwarg will continue to work. You do not break your API. And it gives users (including your own NEB calculation routines) a clear way to ask your system to use BigChem. You do not need checks for system readiness--if BigChem is not available/ready, it will raise the correct errors so end users know why BigChem is not available. With your current approach all exceptions will get burred in your `CheckBigChem` function and user will get no feedback on why it's not available, or worse, their code will simply hang unexpectedly for hours if there is lots of work on the queue and they won't know why.

Reply:
If you must have a check for the BigChem system (e.g., in your tests so they skip or something), you could use this, which checks to see if your client code can connect to BigChem _without_ relying on doing a calculation

```python
from bigchem.app import bigchem as bigchem_app


def bigchem_available(max_retries: int = 2):
    try:
        # Use the connection() method to get a broker connection object
        with bigchem_app.connection() as conn:
            # Open a connection to the broker
            conn.ensure_connection(max_retries=max_retries)
            print("Successfully connected to BigChem.")
            return True
    except Exception as e:
        print(f"Failed to connect to the broker: {str(e)}")
        return False


# Perform the check
bigchem_available(0)

```

You can use `max_retries=0` for an instant check, or increase `max_retries` for most robust checks.

Reply:
@hjnpark tagging this comment as I left the above as a comment instead of a review and I don't want you to miss it!

Reply:
This is exactly what I needed. I had to check whether BigChem is working or not for the tests. I will revise my code based on your comments. Thank you!